pyiron_base/state/settings.py

```diff
@@ -0,0 +1,394 @@
+
+        if os.path.isfile(config_file):
+            parser = ConfigParser(inline_comment_prefixes=(";",))
+            parser.read(config_file)
+            config = {}
+            for sec_name, section in parser.items():
+                for k, v in section.items():
+                    try:
+                        config[self.file_configuration_map[k.upper()]] = v
+                    except KeyError:
+                        pass
+        else:
+            config = None
+
+        return config
+
+    def _update_from_dict(self, config: Dict, map_: Union[None, Dict] = None) -> None:
+        """
+        Overwrite values of the configuration dictionary based on a new dictionary.
+
+        Non-string non-None items are converted to the expected type and paths are converted to absolute POSIX paths.
+        """
+        self._validate_sql_configuration(config=config)
+        self._validate_viewer_configuration(config=config)
+
+        for key, value in config.items():
+            key = key if map_ is None else map_[key]
+
+            if key in ["resource_paths", "project_paths"]:
+                self._configuration[key] = self._convert_to_list_of_paths(
+                    value,
+                    ensure_ends_with="/" if key == "project_paths" else None
+                )
+            elif key == "connection_timeout":
+                self._configuration[key] = int(value)
+            elif key == "sql_file":
+                self._configuration[key] = self.convert_path_to_abs_posix(value)
+            elif key in ["project_check_enabled", "disable_database"]:
+                self._configuration[key] = value if isinstance(value, bool) else strtobool(value)
+            elif key not in self._configuration.keys():
+                raise KeyError(
+                    f"Got unexpected configuration key {key}, please choose from among {self._configuration.keys()}"
+                )
+            else:
+                self._configuration[key] = value
+
+    def _convert_to_list_of_paths(
+            self, paths: Union[str, List[str]], ensure_ends_with: Union[None, str] = None
+    ) -> List[str]:
+        if isinstance(paths, str):
+            paths = paths.replace(',', os.pathsep).split(os.pathsep)
+        return [
+            self.convert_path_to_abs_posix(p)
+            if ensure_ends_with is None or self.convert_path_to_abs_posix(p).endswith(ensure_ends_with)
+            else self.convert_path_to_abs_posix(p) + ensure_ends_with
+            for p in paths
+        ]
+
+    @property
+    # @deprecate("Use pyiron_base.state.state.logger")
+    def logger(self):
+        return logger
+
+    @property
+    # @deprecate("Use pyiron_base.state.state.queue_adapter")
+    def queue_adapter(self):
+        from pyiron_base.state import state
+        return state.queue_adapter
+
+    @property
+    # @deprecate("Use pyiron_base.state.state.publications.list()")
+    def publication_lst(self):
```

Comment:
😞 As long as we need this for our backwards compatibility, we cannot un-singleton the publications...